src/main/java/org/folio/circulation/resources/PickSlipsReportResource.java

```diff
@@ -0,0 +1,153 @@
+
+  /**
+   * The optimal number of identifiers that will not exceed the permissible length
+   * of the URI according to the RFC 2616
+   */
+  private static final int UUID_BATCH_SIZE = 40;
+
+  private static final String SERVICE_POINT_ID_PARAM = "servicePointId";
+  private static final String STATUS_NAME_KEY = "status.name";
+  private static final String REQUEST_STATUS_KEY = "status";
+  private static final String REQUEST_ITEM_ID_KEY = "itemId";
+  private static final String REQUESTS_KEY = "requests";
+  private static final String ITEMS_KEY = "items";
+  private static final String TOTAL_RECORDS_KEY = "totalRecords";
+
+  private final String rootPath;
+
+  public PickSlipsReportResource(String rootPath, HttpClient client) {
+    super(client);
+    this.rootPath = rootPath;
+  }
+
+  @Override
+  public void register(Router router) {
+    RouteRegistration routeRegistration = new RouteRegistration(rootPath, router);
+    routeRegistration.getMany(this::getMany);
+  }
+
+  private void getMany(RoutingContext routingContext) {
+    final UUID servicePointId = UUID.fromString(routingContext.request().getParam(SERVICE_POINT_ID_PARAM));
+
+    final WebContext context = new WebContext(routingContext);
+    final Clients clients = Clients.create(context, client);
+
+    final CollectionResourceClient requestsStorageClient = clients.requestsStorage();
+    final ReportRepository reportRepository = new ReportRepository(clients);
+    final ItemRepository itemRepository = new ItemRepository(clients, true, false, false);
+
+    reportRepository.getAllItemsByField(STATUS_NAME_KEY, PAGED.getValue())
+        .thenApply(r -> r.next(this::mapFetcherToItems))
+        .thenComposeAsync(r -> r.after(items -> filterRequestedItems(items, requestsStorageClient)))
+        .thenComposeAsync(r -> r.after(items -> allOf(items, itemRepository::fetchItemRelatedRecords)))
+        .thenApply(r -> r.next(items -> filterItemsByServicePoint(items, servicePointId)))
+        .thenApply(r -> r.next(this::mapResultToJson))
+        .thenApply(OkJsonResponseResult::from)
+        .thenAccept(r -> r.writeTo(routingContext.response()));
+  }
+
+  private Result<List<Item>> mapFetcherToItems(ItemsReportFetcher fetcher) {
+    return Result.succeeded(
+        fetcher.getResultListOfItems().stream()
+        .flatMap(resultListOfItem -> resultListOfItem.value().getRecords().stream())
+        .collect(Collectors.toList()));
+  }
+
+  private Result<List<Item>> filterItemsByServicePoint(List<Item> items, UUID servicePointId) {
+    return Result.succeeded(
+        items.stream()
+        .filter(item -> servicePointId.equals(item.getLocation().getPrimaryServicePointId()))
+        .collect(Collectors.toList()));
+  }
+
+  private CompletableFuture<Result<List<Result<Item>>>> filterRequestedItems(
+      List<Item> items, CollectionResourceClient client) {
+
+    List<String> itemIds = items.stream()
+        .map(Item::getItemId)
+        .collect(Collectors.toList());
```

Comment:
```suggestion
        .collect(Collectors.toSet());
```

Maybe to set, or use `distinct()`.